packages/gcf-utils/test/gcf-utils.ts

```diff
@@ -33,311 +36,374 @@ function nockRepoList() {
+      it('returns a new pino-based logger instance on first call', () => {
+        const logger = GCFLogger.get();
+        assert.equal(logger.constructor.name, 'Pino');
+      });
+      it('returns the same logger instance on two consecutive calls', () => {
+        const logger1 = GCFLogger.get();
+        const logger2 = GCFLogger.get();
+        assert.deepEqual(logger1, logger2);
+      });
+    });
 
-      enqueueTask = sinon.stub(bootstrapper, 'enqueueTask');
+    describe('logger instance', () => {
+      let writeStream: ObjectWritableMock;
+      let logger: pino.Logger;
+
+      function readLogsAsObjects(writeStream: ObjectWritableMock): LogLine[] {
+        try {
+          writeStream.end();
+          const lines: string[] = writeStream.data;
+          return lines.map(line => JSON.parse(line));
+        } catch (error) {
+          throw new Error(`Failed to read stream: ${error}`);
+        }
+      }
+
+      function testAllLevels() {
+        const levels: {[index: string]: number} = {
+          trace: 10,
+          debug: 20,
+          info: 30,
+          metric: 30,
+          warn: 40,
+          error: 50,
+        };
+
+        for (const level of Object.keys(levels)) {
+          it(`logs ${level} level string`, () => {
+            logger[level]('hello world');
+            const loggedLines: LogLine[] = readLogsAsObjects(writeStream);
+            validateLogs(loggedLines, 1, ['hello world'], [], levels[level]);
+          });
 
-      handler = await bootstrapper.gcf(async app => {
-        app.auth = () =>
-          new Promise<GitHubAPI>(resolve => {
-            resolve(GitHubAPI());
+          it(`logs ${level} level json`, () => {
+            logger[level]({hello: 'world'});
+            const loggedLines: LogLine[] = readLogsAsObjects(writeStream);
+            validateLogs(loggedLines, 1, [], [{hello: 'world'}], levels[level]);
           });
-        app.on('issues', spy);
-        app.on('schedule.repository', spy);
-        app.on('err', sinon.stub().throws());
+        }
+      }
+
+      beforeEach(() => {
+        writeStream = new ObjectWritableMock();
+        logger = GCFLogger['initLogger'](undefined, writeStream);
       });
-    });
 
-    afterEach(() => {
-      sendStub.reset();
-      sendStatusStub.reset();
-      spy.reset();
-      configStub.reset();
-      enqueueTask.reset();
+      testAllLevels();
     });
+  });
 
-    it('calls the event handler', async () => {
-      req.body = {
-        installation: {id: 1},
-      };
-      req.headers = {};
-      req.headers['x-github-event'] = 'issues';
-      req.headers['x-github-delivery'] = '123';
-      req.headers['x-cloudtasks-taskname'] = 'my-task';
-
-      await handler(req, response);
-
-      sinon.assert.calledOnce(configStub);
-      sinon.assert.notCalled(sendStatusStub);
-      sinon.assert.calledOnce(sendStub);
-      sinon.assert.calledOnce(spy);
-    });
+  describe('GCFBootstrapper', () => {
```

Comment:
Existing tests were moved into the nested describe block `GCFBootstrapper`, hence the high number of changed lines

Reply:
Do these tests need to live in the same test file? Maybe it's worthwhile to rename the `gcf-utils.ts` to `gcf-bootstrapper.ts` and have a separate `gcf-logger.ts` test.

It would also make the diff easier to read.